File: 19th April - Wifi Range/19thApril.cpp
// C++ solution
class Solution{
    public:
    bool wifiRange(int N, string S, int X){
        int l = -1;
        for(int i = 0; i < N; i++)
        {
            if(S[i] == '1')
            {
                if(i-X-1 > l) return false;
                l = i+X;
            }
        }
        if(l < N-1) return false;
        return true;
    }
};

File: 19th April - Wifi Range/19thApril.java
// Java solution
class Solution 
{ 
    boolean wifiRange(int N, String S, int X) 
    { 
        // code here
        int c = 0;
        int i = 0;
        while(i<N){
            if(S.charAt(i)=='0')
                c++;
            else
                c = (-1)*(X);
            i++;
            if(c>X)
                return false;
        }
        if(c>0)
            return false;
        return true;

    }
} 